### DISCUSSION TITLE
Version 4.0.0 stats file wiki

### DISCUSSION THREAD

**Discussion by jayant4001** [rtl-airband/RTLSDR-Airband#288] [Q&A] [answered]
Hi,

Thanks for the update! Just compiled and installed v 4.0.0. The stats file has a bunch of new params and since there seem to be a fair amount of updates to the signal and noise level measurements, the noise figure numbers which I was using in prometheus+grafana has broken. I'd really appreciate an update to the wiki with a bit of detail / explanation about the parameters being logged would be very welcome to fix my dashboard.

If anyone has figured out the new stats and can comment here, that would be helpful too!

Thanks!

**Comment by charlie-foxtrot** [comment, marked as answer]
There are a few changes to signal levels that are not backwards compatible (see release notes here: https://github.com/szpajder/RTLSDR-Airband/releases/tag/v4.0.0)

This will mean a dashboard looking across the time an update occurred will be viewing data at different scales. This should go away once the old data is purged or if you zoom into a period of only new data. 

The prom file itself has a text description of each field, (see the “#HELP…” prints here https://github.com/szpajder/RTLSDR-Airband/blob/master/src/output.cpp#L590-L717). Are theses descriptions what you’re looking for?  They could be copy/pasted into the wiki, but then it will need to be updated with the code.

**Reply by jayant4001** [reply]
Thanks for the reply. I understand that the data is not backwards compatible, but I'm a bit confused with the units - the stats file shows noise figures in fractional decimal values, but the text waterfall shows noise and signal levels in what looks like dbFS units. Is there a way to convert the values written to the stats file into the same units as the waterfall? Or am i missing something?

**Reply by charlie-foxtrot** [reply]
Yes, you are correct.  The stats file is using the raw signal level while the waterfall converts that to dBFS before printing.  The conversion is done by `level_to_dBFS()` here: https://github.com/szpajder/RTLSDR-Airband/blob/master/src/util.cpp#L215-L217

**Reply by jayant4001** [reply]
Thanks for clarifying. I believe it would be great from a dashboard point of view to write the dBFS values to the stats file, so if this could be done in a future version without too much pain, it would be great. I assume the same argument would apply to the signal level stats. I guess it could be done externally, but the dependency on `fft_size` make this a little tricky. 

My coding ability is seriously impaired, so here's a question: is it feasible to convert using `level_to_dBFS()` at the point where the value is written to the file? i.e. in this statement: `fprintf(f, "\t%.3f\n", channel->freqlist[k].squelch.noise_level()`.

**Reply by jayant4001** [reply]
Update: I tried making the change in output.cpp and recompiled it. It works, and outputs the stats in dBFS:
`fprintf(f, "\t%.3f\n", level_to_dBFS(channel->freqlist[k].squelch.noise_level()));`
No significant change in the cpu levels, so I'm assuming this is workable for now.

Thanks for the responses @charlie-foxtrot. Great job on the level and squelch re-work!
